### PR TITLE
Update traces exporter to OpenTelemetry HTTP Trace Exporter

### LINKED ISSUE #96: Update trace provider to standard OpenTelemetry HTTP trace provider
Starting with v0.14, we provide tracing in the Managed Tokens library.  This is accomplished using a Jaeger trace provider [here](https://github.com/fermitools/managed-tokens/blob/865160af65277c6a2f858c5589132ea568a0c599/internal/tracing/tracing.go#L19).  

This decision was made, because currently, the Fermilab monitoring architecture only supports Jaeger traces.  Eventually, that will be upgraded to accept more general OpenTelemetry traces.  At that point, we should update the function containing the line above to use the OpenTelemetry Trace HTTP exporter package [here](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp) and update all the dependencies.

I believe, given the construction of the code, that this should be the only function we need to touch.

### PR DESCRIPTION
**Pull request by shreyb**
Closes #96 

Besides the change in the title, we also change a configuration knob, `tracing.url` to `tracing.endpoint` to be more technically correct.

There appear to be many more changes in this PR than are actually there.  This is because Go's dependency manager, `go modules`, automatically adds to the repository the new dependencies needed (and removes the old ones) for the change from the Jaeger Trace exporter libraries to the OpenTelemetry Trace exporter libraries.

Thus, the `go.mod`, `go.sum`, and any `vendor/*` files can be ignored for this PR, as they are managed automatically by the `go` tools.

The only files to actually review are:

- [ ] cmd/refresh-uids-from-ferry/main.go
- [ ] cmd/refresh-uids-from-ferry/main_test.go
- [ ] cmd/run-onboarding-managed-tokens/main.go
- [ ] cmd/token-push/main.go
- [ ] internal/tracing/tracing.go (where the biggest changes are - the above files just use the changes from this file)
- [ ] managedTokens.yml

### CODE REVIEW

File: managedTokens.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ loki:
 
 # Tracing settings
 tracing:
-  url: hostname.domain # Required for tracing use
+  endpoint: scheme://hostname.domain # Required for tracing use
```
Comment (retzkek) [inline managedTokens.yml:75]: 1) In general I'm not a big fan of breaking config changes, but as the sole deployment it's not a stopper.

2) The endpoint is in fact a URL now, so either really is ok.

Comment (shreyb) [reply managedTokens.yml:75]: Oh that's a good point.  We had the discussion and I forgot to change this back.  I'll do that now across the board.

Comment (shreyb) [reply managedTokens.yml]: Done.

**Comment by retzkek** [comment]
Thanks for the changes. LGTM

**Pull request merged**